Rudder-Braze/Classes/RudderBrazeIntegration.m

```diff
@@ -449,6 +383,47 @@ - (ConnectionMode)getConnectionMode:(NSDictionary *)config {
     }
 }
 
+#pragma mark - Push Notification support
+
+// Refer to the Braze doc: https://www.braze.com/docs/developer_guide/platform_integration_guides/swift/push_notifications/integration/#manual-push-integration
+
+// - Register the device token with Braze
+
+-(void)didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
+    [self->braze.notifications registerDeviceToken:deviceToken];
+    [RSLogger logInfo:@"Braze registerDeviceToken:"];
+}
+
+// - Add support for silent notification
+
+- (void)didReceiveRemoteNotification:(NSDictionary *)userInfo fetchCompletionHandler:(void (^)(UIBackgroundFetchResult))completionHandler {
+    // As asked in the Braze doc: https://www.braze.com/docs/developer_guide/platform_integration_guides/swift/push_notifications/integration/#step-3-enable-push-handling, push integration code is called in application’s main thread.
+    dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
We should not use thread. We should forward it to the API. It is more flexibility to the developer if they want to use the thread or not.!

Reply:
I just wanted to understand/confirm, that even though it is mentioned to call those Push APIs in the main thread, should we pass on the responsibility to the developer?

Reply:
Yes. If we are implementing the thread, then we are restricting the use of the thread. A developer may or may not want to use it.